cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -1074,6 +1074,18 @@
 (decl avx512f_enabled () Type)
 (extern extractor avx512f_enabled avx512f_enabled)
 
+(decl avx512bitalg_enabled () Type)
+(extern extractor avx512bitalg_enabled avx512bitalg_enabled)
+
+(decl use_lzcnt () Type)
+(extern extractor use_lzcnt use_lzcnt)
+
+(decl use_bmi1 () Type)
+(extern extractor use_bmi1 use_bmi1)
+
+(decl use_popcnt () Type)
+(extern extractor use_popcnt use_popcnt)
```

Comment:
It seems weird that these extractors (and above) are in any way related to `Type`... logically confusing, I can see the pragmatic part of it.

Reply:
When I introduced the first of these extractors, I did it this way so that you read them near the type constraints because they are most often related to those, e.g. whether to use AVX512 is related to whether we are lowering a vector type.

/me shrugs

Reply:
Yup, the advantage is that it lets us use these in a context that's already common. Though I wonder now if it could make sense to have an extractor result value (the input) that is something like a unit type: in other words, the extractor takes nothing as input, but can be used in any context. I'll think more about it; it probably muddies the waters a bit wrt clean/simple semantics; but it could work...